contracts/TokenDistributor.sol

```diff
@@ -118,42 +120,48 @@ contract TokenDistributor is ITokenDistributor {
     /// @inheritdoc ITokenDistributor
     function claimEpoch(address to, uint epochNumber) public override {
         _checkpointWhenReady();
-        _claimEpoch(to, epochNumber);
+        uint256 proportionalFees = _claimEpoch(to, epochNumber);
+
+        lastTokenBalance -= proportionalFees;
+
+        rewardsToken.transfer(to, proportionalFees);
     }
 
     /// @notice internal claimEpoch function
-    function _claimEpoch(address to, uint epochNumber) internal {
+    function _claimEpoch(address to, uint epochNumber) internal returns (uint256 proportionalFees) {
         _isEpochReady(epochNumber);
-        mapping(uint256 => bool) storage claimedEpochsTo = claimedEpochs[to];
-        if (claimedEpochsTo[epochNumber]) {
+        if (_claimedEpochsBitMap[to].get(epochNumber)) {
             revert CannotClaimTwice();
         }
-        claimedEpochsTo[epochNumber] = true;
+        _claimedEpochsBitMap[to].set(epochNumber);
 
-        uint256 proportionalFees = calculateEpochFees(to, epochNumber);
+        proportionalFees = calculateEpochFees(to, epochNumber);
 
         if (proportionalFees == 0) {
             revert CannotClaim0Fees();
         }
 
-        lastTokenBalance -= proportionalFees;
-
-        rewardsToken.transfer(to, proportionalFees);
-
         emit EpochClaim(to, epochNumber, proportionalFees);
     }
 
     /// @inheritdoc ITokenDistributor
     function claimMany(address to, uint[] calldata epochs) public {
         _checkpointWhenReady();
         uint256 length = epochs.length;
-        for (uint i = 0; i < length; ) {
+        uint256 totalProportionalFees;
+        for (uint i; i < length; ) {
             uint epochNumber = epochs[i];
-            _claimEpoch(to, epochNumber);
+            // TODO: possible optimization by placing the line below in unchecked block
+            // if we can guarantee `totalProportionalFees` will never overflow
```

Comment:
unlikely to overflow because the rewardsToken will be USDC, but still a possibility for anyone using this contract with an arbitrary ERC20

would just leave as is, probably not worth gas savings

Reply:
I've removed the TODO.